### PR TITLE
active pinpointers finds targets upon map transition

### PR DESCRIPTION
**Pull request by JustinTrotter**
## About the PR 
Fixes: #10161

Adds a new FTL state called Arrived between Arriving and Cooldown. This is necessary due to a race condition when trying to locate the disk as soon as the map loads. Waiting for the Arrived FTL State (3 seconds) allows for the map to load and for the disk to be findable.

**Screenshots**

**Changelog**

:cl:
- add: Activated pinpointers now automatically target the disk after completing a FTL jump.

### CODE REVIEW

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
This should be done on cooldown and shouldn't need a new specific FTL state because the map has already changed.

You're also resolving IEntityManager in a loop which should not be done, plus IEntityManager is already on systems.

**Comment by JustinTrotter** [comment]
> This should be done on cooldown and shouldn't need a new specific FTL state because the map has already changed.

I think I see what you mean. I'll set up a cooldown within the Arriving FTL state with the same time frame and remove the FTL state.

> You're also resolving IEntityManager in a loop which should not be done, plus IEntityManager is already on systems.

I see the issue with not resolving it within a loop and will correct that now. But I don't understand what you mean that IEntityManager is already on systems.

**Comment by mirrorcult** [comment]
entitysystems all already have an `EntityManager` field as well as proxies to the methods on it, in this case `TryComp`

**Comment by JustinTrotter** [comment]
Looks like I was only experiencing the race condition because I was making the event call at the very beginning the Arriving case. Moving it to the end seems to have given it enough time to load in the disk.

**Comment by metalgearsloth** [comment]
> Looks like I was only experiencing the race condition because I was making the event call at the very beginning the Arriving case. Moving it to the end seems to have given it enough time to load in the disk.

You just need to wait until the coordinates are set as then the parent has been updated which is lines 278 / 288.

**Comment by JustinTrotter** [comment]
This is ready btw.

File: Content.Server/Pinpointer/ServerPinpointerSystem.cs [hunk 1]
```diff
@@ -13,12 +14,30 @@ public override void Initialize()
...
+            LocateTarget(uid, component);
+        }
+
+        private void OnLocateTarget(HyperspaceJumpCompletedEvent ev)
+        {
+            // This feels kind of expensive, but it only happens once per hyperspace jump
+            foreach (var uid in ActivePinpointers)
+            {
+                TryComp<PinpointerComponent>(uid, out var component);
+                if (component != null)
```
Comment (metalgearsloth) [inline Content.Server/Pinpointer/ServerPinpointerSystem.cs:32]:
```suggestion
                if (TryComp<PinpointerComponent>(uid, out var component))
```

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**